New systemd::timer define type (#119)
camptocamp/puppet-systemd: README.md

~> Class['systemd::tmpfiles']
```

### timer units
Create a timer unit and a service unit to execute from
that timer at the same time.

```puppet

> systemd::timer{'runoften.timer':

Reviewer: Should the timer be automatically added? It's a bit redundant.

Author: @ekohl I'm not sure what you mean here?

Reviewer: Should it accept `systemd::timer{'runoften':` and in timer.pp add `.timer` internally?

Author: Got you, that `.timer` I went through this myself. I thought it was more consistent with `systemd::unit_file` for instance.

Certainly it would remove all that basename dancing.

Author: I think I prefer it with the .timer forced, it makes it consistent with
the `service_unit` parameter as well.

~> Class['systemd::tmpfiles']
```

### timer units
Create a timer unit and a service unit to execute from
that timer at the same time. The following will manage timer unit 
service and enable it.

```puppet
systemd::timer{'runoften.timer':
  timer_source   => "puppet:///modules/${module_name}/runoften.timer",
  service_source => "puppet:///modules/${module_name}/runoften.service",
  active         => true,
  enable         => true,
}
```
or a trivial daily run and manage the timer unit manually

```puppet
$_timer = @(EOT)
[Timer]
OnCalendar=daily
RandomizedDelaySec=1d
EOT

$_service = @(EOT)
[Service]
Type=oneshot
ExecStart=/usr/bin/touch /tmp/file
EOT

> if versioncmp($facts['puppetversion'],'6.1.0') < 0 {

Reviewer: When using `enable` and/or `active` I think that it already takes care of this, right? Might be good to explain this to the user.

Reviewer 2: Yes, I think so.  This adds a `service` resource and the required (for puppet 5) relationship between daemon reload and the service.

Author: Link added:

```
@see https://www.freedesktop.org/software/systemd/man/systemd.timer.html systemd.timer(5)
```
seems to be the correct syntax.

>   include systemd::systemctl::daemon_reload

Reviewer 2: This will always end up included by the main class (which is included by unit_file).
https://github.com/camptocamp/puppet-systemd/blob/f81339ad10516905e460d2899432ce92d83a35d2/manifests/init.pp#L113

>   Systemd::Timer['daily.timer'] ~> Class[Systemd::Systemctl::Daemon_reload]

Reviewer 2: This relationship already exists??
Should be
```
Class[Systemd::Systemctl::Daemon_reload] -> Service['daily.timer']
```
???

Author: Seems bad to notify things inside the unit_file type , assumes it's contents.

Reviewer 2: Not sure I follow. https://github.com/camptocamp/puppet-systemd/blob/f81339ad10516905e460d2899432ce92d83a35d2/manifests/unit_file.pp#L86 is the relationship I think you're duplicating.

Reviewer 2: Actually, I think it's
```
Systemd::Timer['daily.timer'] ~> Service['daily.timer']
Class[Systemd::Systemctl::Daemon_reload] ~> Service['daily.timer']
```
that would be needed.

Perhaps systemd::timer can always make sure a service resource is declared itself? (which I think would be ok for timer units created with this module)
```puppet
unless $enable or $active {
  # if either $enable or $active are set, unit_file will create a service resource and 
  service { $name:
    provider  => systemd,
    subscribe => [
      Class[Systemd::Systemctl::Daemon_reload],
      Systemd::Unit_file[$name],
    ],
  }
}
```
or
```puppet
service { $name:
  provider  => systemd,
  enable    => $enable,
  active    => $active,
  subscribe => [
    Class[Systemd::Systemctl::Daemon_reload],
    Systemd::Unit_file[$name],
  ],
}
systemd::unit_file{$name:
    # as before but active/enable parameters removed
    ensure    => $ensure,
    content   => $timer_content,
    source    => $timer_source,
    path      => $path,
    owner     => $owner,
    group     => $group,
    mode      => $mode,
    show_diff => $show_diff,
}
```

Author: > Not sure I follow.
was me confused.

Creating the service inside timer.pp seems a shame to not reuse unit_file code.

~> Class['systemd::tmpfiles']
```

### timer units
Create a systemd timer unit and a systemd service unit to execute from
that timer 

The following will create a timer unit and a service unit file.
The execution of `systemctl daemon-reload` will occur.

> As active and enable are set true the puppet service `runoften.timer` will be 

Reviewer: ```suggestion
When `active` and `enable` are set to `true` the puppet service `runoften.timer` will be 
```

> declared and be started and enabled.

Reviewer: ```suggestion
declared, started and enabled.
```

ExecStart=/usr/bin/touch /tmp/file
EOT

systemd::timer{'daily.timer':
  timer_content   => $_timer,
  service_content => $_service,
}

service{'daily.timer':
  ensure => running,

>   notify => Systemd::Timer['daily.timer'],

Reviewer 2: ??? What does this do?

Author: I think it must do time travel.

Changed to subscribe - when the unit changes restart the service.
